### PR TITLE
Pin ruby images with 3.1.3

### PR DESCRIPTION
**Pull request by TonyCTHsu**
## Description

Pin the ruby image with 3.1.3

## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

File: lib-injection/runner/action.yml [hunk 1]
```diff
@@ -22,6 +22,7 @@ runs:
         with:
           repository: 'DataDog/system-tests'
           path: './system-tests'
+          ref: tonycthsu/fix-ruby-images-for-lib-injection-tests
```
Comment (ivoanjo) [inline lib-injection/runner/action.yml:25]: Is this change still needed? E.g. should it be merged to master? (It seems like something for testing...?)

File: lib-injection/build/docker/ruby/dd-lib-ruby-init-test-rails-bundle-deploy/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM ruby:3.1
+FROM ruby:3.1.3
```
Comment (marcotc) [inline lib-injection/build/docker/ruby/dd-lib-ruby-init-test-rails-bundle-deploy/Dockerfile:1]: Why is this being pinned to 3.1.3?

**Review by marcotc — APPROVED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Comment by TonyCTHsu** [comment]
When I was writing the image `FROM ruby:3.1` is with Ruby 3.1.3, but recently 3.1.4 released, so the ruby version ends up different from the Gemfile and lockfile specified. Hence, bundle install is broken.
